bestPractice/Test_BestPracticeTriggerHandler.cls

```diff
@@ -0,0 +1,193 @@
+/**
+ * This class contains unit tests for validating the behavior of Apex classes
+ * and triggers.
+ *
+ * Unit tests are class methods that verify whether a particular piece
+ * of code is working properly. Unit test methods take no arguments,
+ * commit no data to the database, and are flagged with the testMethod
+ * keyword in the method definition.
+ *
+ * All test methods in an organization are executed whenever Apex code is deployed
+ * to a production organization to confirm correctness, ensure code
+ * coverage, and prevent regressions. All Apex classes are
+ * required to have at least 75% code coverage in order to be deployed
+ * to a production organization. In addition, all triggers must have some code coverage.
+ * 
+ * The @isTest class annotation indicates this class only contains test
+ * methods. Classes defined with the @isTest annotation do not count against
+ * the organization size limit for all Apex scripts.
+ *
+ * See the Apex Language Reference for more information about Testing and Code Coverage.
+ */
+
+@isTest
+public class Test_BestPracticeTriggerHandler {
+    static Integer i;
+    
+    // Check with valid data
+    static testMethod void validDataCheck() {
+        List<Account> accountListData = BestPracticeDataFactory.accountData();
+        List<Contact> contactListData = BestPracticeDataFactory.contactData();
+        
+        // Acccount
+        List<Account> accountList = new List<Account>();
+        for (Account accountInstance : accountListData) {
+            accountInstance.Website='www.gmail.com';
+            accountList.add(accountInstance);
+        }
+        insert accountList;
```

Comment:
once the record is inserted we need to do assert check on its ids so that we are sure that all records are inserted into database.
Eg:
```
// Verify all account records inserted into database
System.assertEquals(200, accountListData.size())
for (Account accountInstance : accountListData) {
    System.assert(accountInstance.Id != null);
}
```